flink-core/src/main/java/org/apache/flink/api/common/ArchivedExecutionConfig.java

```diff
@@ -54,6 +54,19 @@ public ArchivedExecutionConfig(ExecutionConfig ec) {
 		}
 	}
 
+	public ArchivedExecutionConfig(
+			String executionMode,
+			String restartStrategyDescription,
+			int parallelism,
+			boolean objectReuseEnabled,
+			Map<String, String> globalJobParameters) {
+		this.executionMode = executionMode;
```

Comment:
The checks will only throw exceptions if the current behaviour changes, most probably because of an unintended change. Anyways, feel free to leave it as is.